app/models/framework_response/collection.rb

```diff
@@ -13,6 +14,8 @@ def value=(raw_value)
       self.value_json =
         if response_details && collection_type_valid?(raw_value)
           details_collection(raw_value).to_a
+        elsif multiple_items? && collection_type_valid?(raw_value)
```

Comment:
I intend to resolve the type validation and raise an error before going into this for all response types, as this is not ideal to check the type at every point. Will open a pr after all this is merged